src/ray/common/scheduling/scheduling_ids.cc

```diff
@@ -10,16 +10,16 @@ int64_t StringIdMap::get(const string sid) {
   }
 };
 
-int64_t StringIdMap::insert(const string sid, bool test) {
+int64_t StringIdMap::insert(const string& sid, bool test) {
   auto sit = string_to_int_.find(sid);
   if (sit == string_to_int_.end()) {
     int64_t id = test ? hasher_(sid) % 10 : hasher_(sid);
     for (int i = 0; true; i++) {
       auto it = int_to_string_.find(id);
       if (it == int_to_string_.end()) {
         /// No hash collision, so associated sid with id.
-        string_to_int_.insert(make_pair(sid, id));
-        int_to_string_.insert(make_pair(id, sid));
+        string_to_int_.emplace(sid, id);
+        int_to_string_.emplace(id, sid);
         break;
       }
       id = test ? hasher_(sid + to_string(i)) % 10 : hasher_(sid + to_string(i));
```

Comment:
I know this isn't part of this PR, but this line is pretty unreadable. Something like this seems easier to follow:
```
id = hasher_(sid + to_string(i));
if (test) {
  id = id % 10;
}
```

Reply:
Also it's probably nicer to have a `int64_t max_id=-1` parameter instead of exposing that this is used for testing and hardcoding `10` between here and the tests.

Reply:
Let me replace the 10 with a constant.

Reply:
I guess as discussed we want to get rid of this part of the code anyways and replace it with `ResourceId`/`NodeId` class that caches the hash.

Reply:
Done